### PR TITLE
JS: Safari 9 support

### LINKED ISSUE #2265: Splore in Safari crashes with "TypeError: Invalid attempt to destructure non-iterable instance"
```
TypeError: Invalid attempt to destructure non-iterable instance
uout.js:13:10384
onloadendout.js:13:9380
```
The code is
```
return (0, f.invariant)(u instanceof d.OrderedSequenceCursor), (0, f.invariant)(c instanceof d.OrderedSequenceCursor), y.next = 9, s.default.awrap(a(!1, u, c));
```
which in un-obfuscated land I'm pretty sure is https://github.com/attic-labs/noms/blob/master/js/src/ordered-sequence-diff.js#L81 i.e.
```
[aHasMore, bHasMore] = await doFastForward(false, aParent, bParent);
```
wuh? @arv

### PR DESCRIPTION
**Pull request by arv**
- Lower case header names in fetch function
- fetchUint8Array should return an Uint8Array and not an ArrayBuffer
- Safari does not have slice on typed arrays

Fixes #2265

### CODE REVIEW

**Comment by arv** [comment]
@kalman Merge if LGTY

**Comment by kalman** [comment]
lgtm

**Comment by kalman** [comment]
flow isn't happy though :-(

**Comment by arv** [comment]
Oh. I'll look at it next week.

**Comment by arv** [comment]
I got some minutes and now Flow and tests pass locally.

File: js/src/browser/bytes.js [hunk 1]
```diff
@@ -57,7 +57,10 @@ export function copy(source: Uint8Array, target: Uint8Array, targetStart: number
 }
 
 export function slice(buff: Uint8Array, start: number, end: number): Uint8Array {
-  return buff.slice(start, end);
+  // Safari does not have slice on typed arrays.
```
Comment (kalman) [inline js/src/browser/bytes.js]: Did `slice` copy? Obviously this new code does. My point being, is it worth changing this code to:
```
if (buff.slice) {
  return buff.slice(start, end);
}
// Safari does not have slice on typed arrays.
const v = ...;
return v;
```

**Comment by kalman** [comment]
lgtm

File: js/src/browser/bytes.js (hunk 1 above)
Comment (arv) [inline js/src/browser/bytes.js]: Yeah, slice copies. There is even a test for it.

Comment (arv) [inline js/src/browser/bytes.js]: But now that I look at the spec we can use `ArrayBuffer.prototype.slice` which does work in Safari 9.

**Pull request merged**